### ISSUE TITLE
E2E test happy path

### ISSUE THREAD

**Issue by phillebaba** [defenseunicorns/zarf#2477]
### Describe what should be investigated or refactored

The current e2e tests are large and require a certain amount of knowledge of Zarf and its code base to run. After spending some time getting the tests to run as a new Zarf user I have identified some friction points which could easily be addressed.

1. The dependencies documentation does not specify that Docker and buildx are required dependencies. While it could be assumed that most people will have Docker installed, they may not have buildx installed.

https://github.com/defenseunicorns/zarf/blob/0fa217351e6b75e4f5b18df6e3096797ee767b8c/site/src/content/docs/contribute/testing.mdx?plain=1#L13-L15

2. The tests assume, unless run in appliance mode that access to a Kubernetes cluster is properly configured. A check exists but it does not seem to do anything as it does not check the error. This results in errors much later, which wastes time and becomes harder to decipher. 

https://github.com/defenseunicorns/zarf/blob/0fa217351e6b75e4f5b18df6e3096797ee767b8c/src/test/common.go#L57-L62

3. It would be nice to have a minimal test run documented to make it easy to verify that the e2e tests can run, as it would take a long time for all the tests to run completely.

4. A lot of these tests seem to have little impact on each other, even those that are running against a cluster. While some test cases are marked to run in parallel some are not. This should probably be explored further to speed up test runs.

5. It could be good to move the test images from Docker Hub to GHCR to reduce the risk of rate limiting occurring when running the e2e locally multiple times in a day.

**Comment by bm54cloud** [comment]
Please assign this to me 😄

**Comment by brandtkeller** [comment]
To restate the intent and scope for this issue - there is a little bit of tribal knowledge currently on running the full test suite locally - to include the E2E tests. 

- What local tooling and dependencies are required
- What expectations for infrastructure exist (IE E2E testing assumes a cluster is available when not in appliance mode)
- What other gaps exist in the documentation

Points 4&5 are potential extensions of scope that we can explore further in follow-on issue(s). Some progress has been made here already and parallelizing tests is something we should change in isolation from other changes.

**Comment by bm54cloud** [comment]
Started draft PR but since I'm only working on this about 4 hours per week, it will take me a while to get through all the docs and validate testing. Will remove draft status once ready for review. https://github.com/zarf-dev/zarf/pull/4834
